src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -176,6 +177,14 @@ public function set($id, $service)
         if (null === $service) {
             unset($this->services[$id]);
         }
+
+        if (isset($this->privates[$id])) {
+            if (null === $service) {
+                @trigger_error(sprintf('Unsetting the "%s" private service from outside of the container is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0. An exception will be thrown instead in Symfony 4.0.', $id), E_USER_DEPRECATED);
+            } else {
+                @trigger_error(sprintf('Resetting the "%s" private service from outside of the container is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0. An exception will be thrown instead in Symfony 4.0.', $id), E_USER_DEPRECATED);
+            }
```

Comment:
``set('doctrine.orm.entity_manager', null)`` is used by the Doctrine manager registry (in the bridge) to reset the entity manager service (due to the fact that the entity manager becomes unusable after an exception during a flush, because the unit of work is know to be in a broken state).
So we should still at least support this use case.
Allowing to reset an existing instance of the ORM cannot happen without a major change in the ORM (which I will ask to be done in Doctrine 3, but this is not planned to happen soon), so we need to be able to reset the service, to be able to use the ORM in long running scripts like a RabbitMQ consumer (if we cannot reset the service, the only solution would be to kill the process)